File: src/features/typing/data/go/longest_increasing_subsequence.go
package main

func lengthOfLIS(nums []int) int {
	tails := []int{}
	for _, num := range nums {
		left, right := 0, len(tails)
		for left < right {
			mid := left + (right-left)/2
			if tails[mid] < num {
				left = mid + 1
			} else {
				right = mid
			}
		}
		if left == len(tails) {
			tails = append(tails, num)
		} else {
			tails[left] = num
		}
	}
	return len(tails)
}

File: src/features/typing/data/dart/longest_increasing_subsequence.dart
class Solution {
  int lengthOfLIS(List<int> nums) {
    final tails = <int>[];
    for (final num in nums) {
      var left = 0;
      var right = tails.length;
      while (left < right) {
        final mid = left + (right - left) ~/ 2;
        if (tails[mid] < num) {
          left = mid + 1;
        } else {
          right = mid;
        }
      }
      if (left == tails.length) {
        tails.add(num);
      } else {
        tails[left] = num;
      }
    }
    return tails.length;
  }
}
